<launch>
  <arg name="map_size_x" default="90.0"/>
  <arg name="map_size_y" default="90.0"/>
  <arg name="map_size_z" default="5.0"/>
  <arg name="resolution" default="0.1"/>
  <arg name="map_type"   default="2"/>
  <!-- generate map -->
  <node pkg="map_generator" type="mockamap_node" name="mockamap_node" output="screen">
    <!-- <rosparam file="$(find simulator)/config/mockamap.yaml"/> -->
    <param name="seed" type="int" value="511"/>
    <param name="update_freq" type="double" value="1.0"/>

    <!--  box edge length, unit meter-->
    <param name="resolution" type="double" value="$(arg resolution)"/>

    <!-- map size unit meter-->
    <param name="x_length" type="int" value="$(arg map_size_x)"/>
    <param name="y_length" type="int" value="$(arg map_size_y)"/>
    <param name="z_length" type="int" value="$(arg map_size_z)"/>
    <!-- 1 perlin noise 3D -->
    <!-- 2 perlin box random map-->
    <!-- 3 2d maze still developing-->
    <param name="type" type="int" value="$(arg map_type)"/>

<!-- 1 perlin noise parameters -->
    <!-- complexity:    base noise frequency,
                        large value will be complex
                        typical 0.0 ~ 0.5 -->
    <!-- fill:          infill persentage
                        typical: 0.4 ~ 0.0 -->
    <!-- fractal:       large value will have more detail-->
    <!-- attenuation:   for fractal attenuation
                        typical: 0.0 ~ 0.5 -->

    <param name="complexity"    type="double" value="0.03"/>
    <param name="fill"          type="double" value="0.3"/>
    <param name="fractal"       type="int"    value="1"/>
    <param name="attenuation"   type="double" value="0.1"/>

<!--    <param name="type" type="int" value="2"/>-->
    <param name="width_min"         type="double" value="0.6"/>
    <param name="width_max"         type="double" value="1.5"/>
    <param name="obstacle_number"   type="int"    value="50"/>
    <param name="height_min"        type="double" value="3.5"/>
    <param name="height_max"        type="double" value="4.5"/>

<!--    <param name="type" type="int" value="3"/>-->
    <param name="road_width"        type="double" value="0.5"/>
    <param name="add_wall_x"        type="int"    value="0"/>
    <param name="add_wall_y"        type="int"    value="1"/>
    <!--    maze type: -->
    <!--    1 recursive division maze-->
    <param name="maze_type"         type="int"    value="1"/>

<!-- 4 maze 3d -->
    <param name="numNodes"        type="int"    value="40"/>
    <param name="connectivity"    type="double" value="0.8"/>
    <param name="nodeRad"         type="int"    value="1"/>
    <param name="roadRad"         type="int"    value="10"/>
    <remap from="/mock_map"  to="/map_generator/global_cloud"/>
  </node>

  <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find mockamap)/rviz/mockamap.rviz" required="true" /> -->

</launch>